text/0225-configuration.md

```diff
@@ -0,0 +1,411 @@
+
+### ParseAndValidateConfigurationFromFile(filepath, format) -> config
+
+An API called `ParseAndValidateConfigurationFromFile` receives a string parameter indicating the file path containing the configuration to be parsed. An optional format parameter may be provided to indicate the format that this configuration uses. YAML *SHOULD* be supported, but JSON, which is a strict subset of YAML, *MAY* be supported if a language ecosystem has good reason. The method returns a `Configuration` model that has been validated. This API *MAY* return an error or raise an exception, whichever is idiomatic to the implementation for the following reasons:
+
+* file doesn't exist or is invalid
+* configuration parsed is invalid
+
+#### Python ParseAndValidateConfigurationFromFile example
+
+```python
+
+filename = "./config.yaml"
+
+try:
+  cfg = opentelemetry.ParseAndValidateConfigurationFromFile(filename)
+except Exception as e:
+  print(e)
+
+filename = "./config.json"
+
+try:
+  cfg = opentelemetry.ParseAndValidateConfigurationFromFile(filename, format="json")
+except Exception as e:
+  raise e
+
+```
+
+#### Go ParseAndValidateConfigurationFromFile example
+
+```go
+
+filename := "./config.yaml"
+cfg, err := otel.ParseAndValidateConfigurationFromFile(filename)
+if err != nil {
+  return err
+}
+
+filename := "./config.json"
+cfg, err := otel.ParseAndValidateConfigurationFromFile(filename, otelconfig.WithFormat("json"))
+if err != nil {
+  return err
+}
+
+```
+
+Implementations *MUST* allow users to specify an environment variable to set the configuration file. This gives flexibility to end users of implementations that do not support command line arguments. Some possible names for this variable:
+
+* `OTEL_CONFIGURATION_FILE`
+* `OTEL_CONFIG_FILE`
+* `OTEL_CONF`
+
+### Configuration model
+
+To allow SDKs and instrumentation libraries to accept configuration without having to implement the parsing logic, a `Configuration` model *MAY* be provided by implementations. This object:
+
+* has already been parsed from a file or data structure
+* is structurally valid (errors may yet occur when SDK or instrumentation interprets the object)
+
+### Additional interface: ParseAndValidateConfiguration
+
+Each language implementation supporting OpenTelemetry *MAY* support parsing a data structure instead of a file to produce a model. This allows implementations to provide a configuration interface without the expectation on users to parse a configuration file. The following demonstrates how Python and Go may provide a configuration interface to accomplish this:
+
+#### Python ParseAndValidateConfiguration example
+
+```python
+opentelemetry.ParseAndValidateConfiguration(
+    {
+        "scheme_version": "0.0.1",
```

Comment:
Worth calling out explicitly this is not to be confused with the spec schema version?

Reply:
Are you thinking this should be called out in the otep or in the specification?

Reply:
TBH, it could probably just be called "version".

Reply:
It is called `file_format` in Schema Files: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/file_format_v1.1.0.md

Reply:
➕ to `file_format` for consistency.

Reply:
Renamed to file_format.